### PR TITLE
Fix carousel animation crash

### LINKED ISSUE #1878: Carousel is broken.
The carousel in ControlCatalog is broken on master. When clicking the "next" button I get:

```
System.InvalidCastException
  HResult=0x80004002
  Message=Unable to cast object of type 'System.Int32' to type 'System.Double'.
  Source=Avalonia.Animation
  StackTrace:
   at Avalonia.Animation.AnimationInstance`1.DoPlayStatesAndTime(TimeSpan systemTime) in D:\projects\AvaloniaUI\Avalonia\src\Avalonia.Animation\AnimationInstance`1.cs:line 143
```

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #1878

### CODE REVIEW

**Comment by jmacato** [comment]
Hmm this is good but i think we should handle the type conversion from ints to double on the core iteration code just in case some user forgets to do so :)

**Comment by grokys** [comment]
Note that even after this is fixed, the animation seems to be wrong:



It's like the previous image is moving in the opposite direction to the subsequent image or something?

**Comment by jmacato** [comment]
The effect looks cool haha :smile: but there must be some flipped signs somewhere

File: src/Avalonia.Visuals/Animation/PageSlide.cs [hunk 1]
```diff
@@ -86,7 +86,7 @@ public async Task Start(Visual from, Visual to, bool forward)
                         new Setter
                         {
                             Property = translateProperty,
-                            Value = 0
+                            Value = 0.0
```
Comment (wdcossey) [inline src/Avalonia.Visuals/Animation/PageSlide.cs]: **0d** would be better imho.

**Review by jmacato — APPROVED** [review]

**Pull request merged**